tests/Opus/Search/Common/IndexerTest.php

```diff
@@ -0,0 +1,961 @@
+
+		$this->files_dir = Opus_Config::get()->workspacePath . '/files';
+
+		$this->indexer = Opus_Search_Service::selectIndexingService();
+
+		$document = new Opus_Document();
+		foreach ( self::$_validDocumentData as $fieldname => $value ) {
+			$callname = 'set' . $fieldname;
+			$document->$callname( $value );
+		}
+		$document->store();
+		$this->document_id = $document->getId();
+	}
+
+	/**
+	 * This method is called after a test is executed.
+	 */
+	protected function tearDown() {
+		// remove test documents under tests/workspace/files/$document_id
+		$dirname = $this->files_dir . DIRECTORY_SEPARATOR . $this->document_id;
+		if ( is_dir( $dirname ) && is_readable( $dirname ) ) {
+			foreach ( glob( $dirname . "/*" ) as $filename ) {
+				if ( is_readable( $filename ) ) {
+					unlink( $filename );
+				}
+			}
+			rmdir( $dirname );
+		}
+
+		parent::tearDown();
+	}
+
+	/**
+	 * @expectedException Opus_Search_InvalidConfigurationException
+	 */
+	public function testMissingConfigParamSearchEngine_Index_Host() {
+		$this->markTestSkipped( 'not supported in Opus_Search API due to passing configuration to 3rd-party library for processing' );
+
+		$this->adjustConfiguration( array(), function ( Zend_Config $config ) {
+			unset(
+				$config->searchengine->solr->default->service->index->endpoint->primary->host,
+				$config->searchengine->solr->default->service->default->endpoint->primary->host
+			);
+
+			return $config;
+		} );
+
+		Opus_Search_Service::selectIndexingService()
+		                   ->removeAllDocumentsFromIndex();
+	}
+
+	/**
+	 * @expectedException Opus_Search_InvalidConfigurationException
+	 */
+	public function testMissingConfigParamSearchEngine_Index_Port() {
+		$this->markTestSkipped( 'not supported in Opus_Search API due to passing configuration to 3rd-party library for processing' );
+
+		$this->adjustConfiguration( array(), function ( Zend_Config $config ) {
+			unset(
+				$config->searchengine->solr->default->service->index->endpoint->primary->port,
+				$config->searchengine->solr->default->service->default->endpoint->primary->port
+			);
+
+			return $config;
+		} );
+
+		Opus_Search_Service::selectIndexingService()
+		                   ->removeAllDocumentsFromIndex();
+	}
+
+	/**
+	 * @expectedException Opus_Search_InvalidConfigurationException
+	 */
+	public function testMissingConfigParamSearchEngine_Index_Path() {
+		$this->markTestSkipped( 'not supported in Opus_Search API due to passing configuration to 3rd-party library for processing' );
+
+		$this->adjustConfiguration( array(), function ( Zend_Config $config ) {
+			unset(
+				$config->searchengine->solr->default->service->index->endpoint->primary->path,
+				$config->searchengine->solr->default->service->default->endpoint->primary->path
+			);
+
+			return $config;
+		} );
+
+		Opus_Search_Service::selectIndexingService()
+		                   ->removeAllDocumentsFromIndex();
+	}
+
+	public function testMissingConfigParamLogPrepareXml() {
```

Comment:
Any idea what this test does? If not, don't spend time figuring it out. This problem is not new. I will put the test on my todo list. There is no assert, no check of any expectation and no explanation. (Bad test.) I would love to tag it with a TODO, but there the workflow gets tricky again. Can't do it without merging the pull request. Hmm.